### PR TITLE
Add 'run-experiment' script, simpler curriculum config

### PR DESCRIPTION
**Pull request by harperj**
This change adds a new 'mlagents-run-experiment' endpoint which
accepts a single YAML/JSON file providing all of the information that
mlagents-learn accepts via command-line arguments and file inputs.
As part of this change the curriculum configuration is simplified to
accept only a single file for all the curricula in an environment
rather than a file for each behavior.

### CODE REVIEW

File: ml-agents/mlagents/trainers/learn.py [hunk 1]
```diff
@@ -25,35 +25,194 @@
...
-    sampler_file_path: Optional[str]
-    docker_target_name: Optional[str]
-    env_args: Optional[List[str]]
-    cpu: bool
-    width: int
-    height: int
-    quality_level: int
-    time_scale: float
-    target_frame_rate: int
+parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
```
Comment (harperj) [inline ml-agents/mlagents/trainers/learn.py]: Note the parser is defined at the module level now -- this was needed in order to re-use the defaults in the `RunOptions` definition.

File: docs/Migrating.md [hunk 2]
```diff
@@ -11,9 +11,12 @@ The versions can be found in
 
 ### Important changes
 * Trainer steps are now counted per-Agent, not per-environment as in previous versions. For instance, if you have 10 Agents in the scene, 20 environment steps now corresponds to 200 steps as printed in the terminal and in Tensorboard.
+* Curriculum config files are now YAML formatted and all curricula for a training run are combined into a single file.
+* The `--num-runs` command-line option has been removed.
 
 ### Steps to Migrate
 * Multiply `max_steps` and `summary_steps` in your `trainer_config.yaml` by the number of Agents in the scene.
+* Combine curriculum configs into a single file.  See [the WallJump curricula](../config/curricula/wall_jump.yaml) for an example of the new curriculum config format.
```
Comment (chriselion) [inline docs/Migrating.md]: Maybe link to https://www.json2yaml.com/ for help converting?

File: ml-agents/mlagents/trainers/learn.py (hunk 1 above)
Comment (chriselion) [reply ml-agents/mlagents/trainers/learn.py]: Maybe move the setup to a _create_parser() function?

**Comment by chriselion** [comment]
Do you want to have any methods (or examples) for serializing out RunOptions? How are you supposed to produce the input file for `mlagents-run-experiment`? (looks like `RunOptions._asdict()` is want you want https://stackoverflow.com/a/26180604/224264)

File: config/curricula/wall_jump.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+BigWallJump:
+  measure: progress
+  thresholds: [0.1, 0.3, 0.5]
+  min_lesson_length: 100
+  signal_smoothing: true
+  parameters:
+    big_wall_min_height: [0.0, 4.0, 6.0, 8.0]
+    big_wall_max_height: [4.0, 7.0, 8.0, 8.0]
+SmallWallJump:
```
Comment (ervteng) [inline config/curricula/wall_jump.yaml]: Space between this to keep it consistent with `trainer_config.yaml`?

File: ml-agents/mlagents/trainers/learn.py (hunk 1 above)
Comment (harperj) [reply ml-agents/mlagents/trainers/learn.py]: Done

File: config/curricula/wall_jump.yaml (hunk 3 above)
Comment (harperj) [reply config/curricula/wall_jump.yaml]: Sure, done

**Comment by harperj** [comment]
@chriselion I've added a pretty-printed version of the RunOptions when `--debug` is used, and updated based on other comments.

**Pull request merged**